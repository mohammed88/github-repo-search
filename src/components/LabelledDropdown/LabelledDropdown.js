import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Dropdown from '../../ui/Dropdown/';
import {Text12} from '../../ui/styles/typography';
import Spacing from '../../ui/styles/spacing';

const Label = styled.div`
	margin-bottom: ${Spacing.small};
	${Text12};
`;

function LabelledDropdown({className, label, initialValue, options, onChange}) {
	return (
		<div className={className}>
			<Label>{label}</Label>
			<Dropdown list={options} onChange={onChange} initialValue={initialValue} />
		</div>
	);
}

LabelledDropdown.propTypes = {
	label: PropTypes.string.isRequired,
	options: PropTypes.array.isRequired,
	onChange: PropTypes.func,
}

LabelledDropdown.defaultProps = {
	onChange: () => {},
	initialValue: '',
}

export default LabelledDropdown;