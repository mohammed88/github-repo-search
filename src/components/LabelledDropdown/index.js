import LabelledDropdown from './LabelledDropdown';

export default LabelledDropdown;