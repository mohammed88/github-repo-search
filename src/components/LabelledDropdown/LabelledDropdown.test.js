import React from 'react';
import {shallow} from 'enzyme';

import LabelledDropdown from './LabelledDropdown';

describe('LabelledDropdown', () => {
	it('should render', () => {
		const wrapper = shallow(<LabelledDropdown label="Label" options={[]}/>);

		expect(wrapper).toMatchSnapshot();
	});
});