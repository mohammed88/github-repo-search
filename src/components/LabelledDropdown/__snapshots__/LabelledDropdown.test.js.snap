// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelledDropdown should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LabelledDropdown
    initialValue=""
    label="Label"
    onChange={[Function]}
    options={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Label
        </ForwardRef>,
        <Dropdown
          initialValue=""
          list={Array []}
          onChange={[Function]}
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Label",
        },
        "ref": null,
        "rendered": "Label",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
	margin-bottom: ",
              "8px",
              ";
	",
              "
	font-family: ",
              "'Roboto', sans-serif",
              ";
	font-size: 12px;
",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "initialValue": "",
          "list": Array [],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Label
          </ForwardRef>,
          <Dropdown
            initialValue=""
            list={Array []}
            onChange={[Function]}
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Label",
          },
          "ref": null,
          "rendered": "Label",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
	margin-bottom: ",
                "8px",
                ";
	",
                "
	font-family: ",
                "'Roboto', sans-serif",
                ";
	font-size: 12px;
",
                ";
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "initialValue": "",
            "list": Array [],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
